File: src/main/java/pl/training/workshop/m6/s03_typecode/step3/FormatCodes.java
package pl.training.workshop.m6.s03_typecode.step3;

import java.util.EnumMap;
import java.util.Map;

/**
 * Krok 3: mapper migracyjny int &lt;-&gt; Format. Baza i CSV zostają przy kodzie int;
 * gdy kiedyś przejdziemy na kody tekstowe, zmieni się tylko ta klasa.
 */
public final class FormatCodes {
    private static final Map<Format, Integer> CODES = new EnumMap<>(Map.of(
            Format.TWO_D, 1,
            Format.THREE_D, 2,
            Format.IMAX, 3));

    private FormatCodes() {
    }

    public static Format fromCode(int code) {
        for (Map.Entry<Format, Integer> entry : CODES.entrySet()) {
            if (entry.getValue() == code) {
                return entry.getKey();
            }
        }
        throw new IllegalArgumentException("unknown format code: " + code);
    }

    public static int toCode(Format format) {
        return CODES.get(format);
    }
}

File: csharp/src/Training.Workshop/M6/S03TypeCode/Step3/FormatCodes.cs
namespace Training.Workshop.M6.S03TypeCode.Step3;

/// <summary>
/// Krok 3: mapper migracyjny int &lt;-&gt; Format. Baza i CSV zostają przy kodzie int;
/// gdy kiedyś przejdziemy na kody tekstowe, zmieni się tylko ta klasa.
/// </summary>
public static class FormatCodes
{
    private static readonly IReadOnlyDictionary<Format, int> Codes = new Dictionary<Format, int>
    {
        [Format.TwoD] = 1,
        [Format.ThreeD] = 2,
        [Format.Imax] = 3,
    };

    public static Format FromCode(int code)
    {
        foreach (var entry in Codes)
        {
            if (entry.Value == code)
            {
                return entry.Key;
            }
        }
        throw new ArgumentException("unknown format code: " + code);
    }

    public static int ToCode(Format format)
    {
        return Codes[format];
    }
}
